File: seed_recipes.py
"""
Seed the database with sample sushi/poke recipes.
Run: python seed_recipes.py
"""
from recipe_service import create_recipe, get_recipe, ensure_db

SAMPLE_RECIPES = [
    {
        "name": "California Roll",
        "ingredients": [
            {"item": "sushi rice", "quantity": 2000, "unit": "g", "cost_per_unit": 0.004},
            {"item": "crab stick", "quantity": 500, "unit": "g", "cost_per_unit": 0.02},
            {"item": "avocado", "quantity": 5, "unit": "each", "cost_per_unit": 1.50},
            {"item": "cucumber", "quantity": 2, "unit": "each", "cost_per_unit": 0.80},
            {"item": "nori", "quantity": 20, "unit": "sheets", "cost_per_unit": 0.25},
        ],
        "procedures": [
            "Cook and season sushi rice; cool to room temp.",
            "Slice cucumber and avocado into strips.",
            "Place nori on mat, spread rice, add crab, avocado, cucumber, roll tightly.",
            "Slice into 8 pieces per roll.",
        ],
        "supplies": [{"item": "tray", "quantity": 10}, {"item": "lids", "quantity": 50}],
        "base_servings": 50,
    },
    {
        "name": "Spicy Tuna Roll",
        "ingredients": [
            {"item": "sushi rice", "quantity": 2000, "unit": "g", "cost_per_unit": 0.004},
            {"item": "tuna", "quantity": 800, "unit": "g", "cost_per_unit": 0.03},
            {"item": "sriracha", "quantity": 100, "unit": "ml", "cost_per_unit": 0.02},
            {"item": "mayo", "quantity": 150, "unit": "g", "cost_per_unit": 0.01},
            {"item": "nori", "quantity": 20, "unit": "sheets", "cost_per_unit": 0.25},
        ],
        "procedures": [
            "Dice tuna, mix with sriracha and mayo.",
            "Spread rice on nori, add spicy tuna, roll and slice.",
        ],
        "supplies": [{"item": "tray", "quantity": 10}, {"item": "lids", "quantity": 50}],
        "base_servings": 50,
    },
    {
        "name": "Poke Bowl",
        "ingredients": [
            {"item": "sushi rice", "quantity": 2500, "unit": "g", "cost_per_unit": 0.004},
            {"item": "ahi tuna", "quantity": 1200, "unit": "g", "cost_per_unit": 0.035},
            {"item": "soy sauce", "quantity": 200, "unit": "ml", "cost_per_unit": 0.015},
            {"item": "sesame oil", "quantity": 50, "unit": "ml", "cost_per_unit": 0.05},
            {"item": "green onion", "quantity": 3, "unit": "bunch", "cost_per_unit": 1.00},
            {"item": "cucumber", "quantity": 4, "unit": "each", "cost_per_unit": 0.80},
            {"item": "edamame", "quantity": 500, "unit": "g", "cost_per_unit": 0.02},
        ],
        "procedures": [
            "Cook rice and portion into bowls.",
            "Cube tuna, toss with soy and sesame oil.",
            "Chop cucumber and green onion; top bowls with tuna, veggies, edamame.",
        ],
        "supplies": [{"item": "bowl", "quantity": 50}, {"item": "lid", "quantity": 50}],
        "base_servings": 50,
    },
    {
        "name": "Salmon Nigiri",
        "ingredients": [
            {"item": "sushi rice", "quantity": 1800, "unit": "g", "cost_per_unit": 0.004},
            {"item": "salmon", "quantity": 1000, "unit": "g", "cost_per_unit": 0.028},
            {"item": "rice vinegar", "quantity": 80, "unit": "ml", "cost_per_unit": 0.02},
        ],
        "procedures": [
            "Season rice with vinegar, form small oblong portions.",
            "Slice salmon; place slice on each rice portion.",
        ],
        "supplies": [{"item": "tray", "quantity": 15}, {"item": "lids", "quantity": 50}],
        "base_servings": 50,
    },
    {
        "name": "Vegetable Roll",
        "ingredients": [
            {"item": "sushi rice", "quantity": 1500, "unit": "g", "cost_per_unit": 0.004},
            {"item": "cucumber", "quantity": 3, "unit": "each", "cost_per_unit": 0.80},
            {"item": "carrot", "quantity": 4, "unit": "each", "cost_per_unit": 0.50},
            {"item": "avocado", "quantity": 4, "unit": "each", "cost_per_unit": 1.50},
            {"item": "nori", "quantity": 15, "unit": "sheets", "cost_per_unit": 0.25},
        ],
        "procedures": [
            "Julienne cucumber, carrot; slice avocado.",
            "Roll rice and veggies in nori; slice.",
        ],
        "supplies": [{"item": "tray", "quantity": 8}, {"item": "lids", "quantity": 40}],
        "base_servings": 40,
    },
]


def main():
    ensure_db()
    for data in SAMPLE_RECIPES:
        existing = get_recipe(data["name"])
        if existing:
            print(f"Skip (exists): {data['name']}")
            continue
        create_recipe(data)
        print(f"Created: {data['name']}")
    print("Seed done.")


if __name__ == "__main__":
    main()

File: scripts/seed-recipes.ts
/**
 * Seed sample recipes and ingredients for Phase 1 testing.
 * Run: npx tsx scripts/seed-recipes.ts
 * Requires: SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY in env
 */

import { createClient } from '@supabase/supabase-js';

const url = process.env.SUPABASE_URL;
const key = process.env.SUPABASE_SERVICE_ROLE_KEY;
if (!url || !key) {
  console.error('Set SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY');
  process.exit(1);
}

const supabase = createClient(url, key);

async function seed() {
  // Ingredients
  const { data: ingredients, error: ingErr } = await supabase
    .from('ingredients')
    .upsert(
      [
        { name: 'Sushi rice', unit: 'cups', cost_per_unit: 0.5 },
        { name: 'Nori', unit: 'sheets', cost_per_unit: 0.15 },
        { name: 'Tuna', unit: 'lb', cost_per_unit: 12 },
        { name: 'Salmon', unit: 'lb', cost_per_unit: 14 },
        { name: 'Cucumber', unit: 'each', cost_per_unit: 0.5 },
        { name: 'Avocado', unit: 'each', cost_per_unit: 1.2 },
        { name: 'Soy sauce', unit: 'oz', cost_per_unit: 0.1 },
        { name: 'Sriracha', unit: 'oz', cost_per_unit: 0.08 },
      ],
      { onConflict: 'name' }
    )
    .select('id, name');

  if (ingErr) {
    console.error('Ingredients error:', ingErr);
    process.exit(1);
  }

  const byName = new Map((ingredients ?? []).map((i) => [i.name, i.id]));

  // Recipes
  const { data: recipes, error: recErr } = await supabase
    .from('recipes')
    .upsert(
      [
        {
          name: 'Spicy Tuna Roll',
          yield_servings: 50,
          procedure: [
            { step: 'Dice tuna, mix with sriracha mayo', role: 'Chef' },
            { step: 'Roll with rice and nori', role: 'Prep' },
            { step: 'Slice and portion', role: 'Pack' },
          ],
        },
        {
          name: 'Salmon Nigiri',
          yield_servings: 40,
          procedure: [
            { step: 'Slice salmon for nigiri', role: 'Chef' },
            { step: 'Form rice balls, top with salmon', role: 'Prep' },
            { step: 'Arrange on tray', role: 'Pack' },
          ],
        },
        {
          name: 'California Roll',
          yield_servings: 48,
          procedure: [
            { step: 'Prep crab mix', role: 'Chef' },
            { step: 'Roll cucumber, avocado, crab', role: 'Prep' },
            { step: 'Slice and pack', role: 'Pack' },
          ],
        },
      ],
      { onConflict: 'name' }
    )
    .select('id, name');

  if (recErr) {
    console.error('Recipes error:', recErr);
    process.exit(1);
  }

  const recipeByName = new Map((recipes ?? []).map((r) => [r.name, r.id]));

  // Recipe items
  const recipeItems = [
    { recipe: 'Spicy Tuna Roll', ingredient: 'Tuna', quantity: 2 },
    { recipe: 'Spicy Tuna Roll', ingredient: 'Sushi rice', quantity: 8 },
    { recipe: 'Spicy Tuna Roll', ingredient: 'Nori', quantity: 25 },
    { recipe: 'Spicy Tuna Roll', ingredient: 'Sriracha', quantity: 4 },
    { recipe: 'Salmon Nigiri', ingredient: 'Salmon', quantity: 1.5 },
    { recipe: 'Salmon Nigiri', ingredient: 'Sushi rice', quantity: 6 },
    { recipe: 'California Roll', ingredient: 'Sushi rice', quantity: 6 },
    { recipe: 'California Roll', ingredient: 'Nori', quantity: 24 },
    { recipe: 'California Roll', ingredient: 'Cucumber', quantity: 4 },
    { recipe: 'California Roll', ingredient: 'Avocado', quantity: 6 },
  ];

  for (const ri of recipeItems) {
    const recipeId = recipeByName.get(ri.recipe);
    const ingredientId = byName.get(ri.ingredient);
    if (recipeId && ingredientId) {
      await supabase.from('recipe_items').upsert(
        { recipe_id: recipeId, ingredient_id: ingredientId, quantity: ri.quantity },
        { onConflict: 'recipe_id,ingredient_id' }
      );
    }
  }

  console.log('Seeded', ingredients?.length ?? 0, 'ingredients,', recipes?.length ?? 0, 'recipes');
}

seed().catch(console.error);
